src/lib.rs

```diff
@@ -120,25 +118,40 @@ impl Pattern {
     ///     .expect("regex is not invalid");
     /// assert!(pattern2.display_matches(&"hello world! aaaaab"));
     /// ```
-    pub fn new_anchored(pattern: &str) -> Result<Self, Error> {
-        let automaton = dense::Builder::new().anchored(true).build(pattern)?;
-        Ok(Pattern { automaton })
+    pub fn new_anchored(pattern: &str) -> Result<Self, BuildError> {
+        let automaton = DFA::new(pattern)?;
+        Ok(Pattern {
+            automaton,
+            anchored: true,
+        })
     }
 }
 
 impl FromStr for Pattern {
-    type Err = Error;
+    type Err = BuildError;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         Self::new(s)
     }
 }
 
-impl<S, A> Pattern<S, A>
-where
-    S: StateID,
-    A: DFA<ID = S>,
-    Self: for<'a> ToMatcher<'a, S>,
-{
+impl<A: Automaton> Pattern<A> {
+    /// Obtains a `matcher` for this pattern.
+    ///
+    /// This conversion is useful when wanting to incrementally feed input (via
+    /// `io::Write`/`fmt::Write` to a matcher). Otherwise, the convenience methods on Pattern
+    /// suffice.
+    pub fn matcher(&self) -> Matcher<&'_ A> {
+        let config = regex_automata::util::start::Config::new().anchored(if self.anchored {
+            Anchored::Yes
+        } else {
+            Anchored::No
+        });
```

Comment:
take it or leave it: why not just replace the `anchored: bool` field on `Pattern` with an `Anchored` enum value, and do this when the  `Pattern` is constructed, instead?